okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -13,430 +13,403 @@
-  private boolean enqueuedClose;
+  /** True if we've enqueued a close frame. No further message frames will be enqueued.  */
+  private var enqueuedClose = false
 
   /**
-   * When executed this will cancel this websocket. This future itself should be canceled if that is
-   * unnecessary because the web socket is already closed or canceled.
+   * When executed this will cancel this web socket. This future itself should be canceled if that
+   * is unnecessary because the web socket is already closed or canceled.
    */
-  private ScheduledFuture<?> cancelFuture;
+  private var cancelFuture: ScheduledFuture<*>? = null
 
-  /** The close code from the peer, or -1 if this web socket has not yet read a close frame. */
-  private int receivedCloseCode = -1;
+  /** The close code from the peer, or -1 if this web socket has not yet read a close frame.  */
+  private var receivedCloseCode = -1
 
-  /** The close reason from the peer, or null if this web socket has not yet read a close frame. */
-  private String receivedCloseReason;
+  /** The close reason from the peer, or null if this web socket has not yet read a close frame.  */
+  private var receivedCloseReason: String? = null
 
-  /** True if this web socket failed and the listener has been notified. */
-  private boolean failed;
+  /** True if this web socket failed and the listener has been notified.  */
+  private var failed = false
 
-  /** Total number of pings sent by this web socket. */
-  private int sentPingCount;
+  /** Total number of pings sent by this web socket.  */
+  private var sentPingCount = 0
 
-  /** Total number of pings received by this web socket. */
-  private int receivedPingCount;
+  /** Total number of pings received by this web socket.  */
+  private var receivedPingCount = 0
 
-  /** Total number of pongs received by this web socket. */
-  private int receivedPongCount;
+  /** Total number of pongs received by this web socket.  */
+  private var receivedPongCount = 0
 
-  /** True if we have sent a ping that is still awaiting a reply. */
-  private boolean awaitingPong;
+  /** True if we have sent a ping that is still awaiting a reply.  */
+  private var awaitingPong = false
 
-  public RealWebSocket(Request request, WebSocketListener listener, Random random,
-      long pingIntervalMillis) {
-    if (!"GET".equals(request.method())) {
-      throw new IllegalArgumentException("Request must be GET: " + request.method());
+  init {
+    require("GET" == originalRequest.method()) {
+      "Request must be GET: ${originalRequest.method()}"
     }
-    this.originalRequest = request;
-    this.listener = listener;
-    this.random = random;
-    this.pingIntervalMillis = pingIntervalMillis;
 
-    byte[] nonce = new byte[16];
-    random.nextBytes(nonce);
-    this.key = ByteString.of(nonce).base64();
+    val nonce = ByteArray(16)
+    random.nextBytes(nonce)
+    this.key = ByteString.of(*nonce).base64()
```

Comment:
You could inline this with something like

`this.key = ByteArray(16).apply { random.nextBytes(this) }.toByteString().base64()`

Reply:
Done. I still want a random byte string thing. Maybe just `ByteString.random(Random(), 16)` should work.

Reply:
From 4 years ago! https://github.com/square/okio/issues/132